ew_agents/knowledge_retrieval.py

```
@@ -0,0 +1,490 @@
+
+# LlamaIndex imports
+try:
+    from llama_index.core import VectorStoreIndex, Document, Settings
+    from llama_index.core.node_parser import SimpleNodeParser
+    from llama_index.core.retrievers import VectorIndexRetriever
+    from llama_index.core.query_engine import RetrieverQueryEngine
+    from llama_index.core.response_synthesizers import ResponseMode
+    from llama_index.embeddings.huggingface import HuggingFaceEmbedding
+except ImportError:
+    # Fallback for older llama-index versions
+    from llama_index import VectorStoreIndex, Document, ServiceContext
```

Ruff (0.12.2)

`llama_index.ServiceContext` imported but unused

Remove unused import: `llama_index.ServiceContext`

(F401)